terraform/projects/app-ecs-instances/instance-user-data.tpl

```diff
@@ -32,10 +32,15 @@ else
   echo "[$(date '+%H:%M:%S %d-%m-%Y')] attach-volume: /dev/$DEVICE is already formatted: $(file -s /dev/$DEVICE)"
 fi
 
-# Allow prometheus container user access to read/write/execute within container
+# Allow prometheus container user access to read/write metrics to persistent volume mount from within a container
 mkdir -p /ecs/prometheus_data
+
+#Cretate prometheus & grant access to prometheus data files. We create the group & match it to the gid
+groupadd --system --gid 65534 prometheus
+useradd --uid 65534 --gid 65534 --system --groups prometheus
```

Comment:
Is the `--groups` option doing nothing here? We are already specifying that it should join gid `65534` which is the prometheus group?

We could also match the order of the above statement so that `--system` comes first just for prettiness 💄

Reply:
Yep, to be fair I just added that in there for readability. We can take this out since the groupadd shows this.